zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -488,17 +488,9 @@ public void loadData() throws IOException, InterruptedException {
         }
 
         // Clean up dead sessions
-        List<Long> deadSessions = new ArrayList<>();
-        for (Long session : zkDb.getSessions()) {
-            if (zkDb.getSessionWithTimeOuts().get(session) == null) {
-                deadSessions.add(session);
-            }
-        }
-
-        for (long session : deadSessions) {
-            // TODO: Is lastProcessedZxid really the best thing to use?
-            killSession(session, zkDb.getDataTreeLastProcessedZxid());
-        }
+        zkDb.getSessions().stream()
```

Comment:
I don't find this change so useful.
Using streams does not speed up things and it usually add overhead.

Why do you want to make this change ? and the other ones ?

Reply:
I am trying to refactor those for loops that just performs basic filtering and collections. I feel those are better accomplished using streams.

Reply:
@eolivelli Please guide me regarding the kind of refactoring expected as part of issue 3415 (https://issues.apache.org/jira/browse/ZOOKEEPER-3415)

Reply:
I like these changes. If performance is not impacted, I think we're fine. They make the code more readable.

Reply:
@eolivelli Streams don't add much overhead, when used properly. They might add some overhead sometimes, but that's not always the case, and even when they do add a slight overhead, the increased reliability of the code, due to its readability/maintainability can be worth the trade.

In this case, it:

1. Makes two loops (`O(n + m)`) into a single loop (`O(n)`),
2. Avoids creating an intermediate data structure entirely (saving 'm' memory units), and
3. It's much nicer to read.

I see no reason to not apply these changes.